metadata-ingestion/docs/sources/dataplex/dataplex_pre.md

```diff
@@ -0,0 +1,401 @@
+
+**Example Configuration:**
+
+```yaml
+source:
+  type: dataplex
+  config:
+    project_ids:
+      - "my-gcp-project"
+
+    # Location for lakes/zones/entities (if using include_entities)
+    location: "us-central1"
+
+    # Location for entries (Universal Catalog) - defaults to "us"
+    # Must be multi-region (us, eu, asia) for system entry groups like @bigquery
+    entries_location: "us" # Default value, can be omitted
+
+    # API selection
+    include_entries: true # Enable Universal Catalog entries (default: true)
+    include_entities: false # Enable lakes/zones entities (default: false)
+
+    # Metadata extraction settings
+    include_schema: true # Enable schema metadata extraction (default: true)
+    include_lineage: true # Enable lineage extraction with automatic retries
+
+    # Lineage retry settings (optional, defaults shown)
+    lineage_max_retries: 3 # Max retry attempts (range: 1-10)
+    lineage_retry_backoff_multiplier: 1.0 # Exponential backoff multiplier (range: 0.1-10.0)
+```
+
+**Advanced Configuration for Large Deployments:**
+
+For deployments with thousands of entities, memory optimization is critical. The connector uses batched emission to keep memory bounded:
+
+```yaml
+source:
+  type: dataplex
+  config:
+    project_ids:
+      - "my-gcp-project"
+    location: "us-central1"
+    entries_location: "us"
+
+    # API selection
+    include_entries: true
+    include_entities: false
+
+    # Memory optimization for large deployments
+    batch_size:
+      1000 # Batch size for metadata emission and lineage extraction
+      # Entries/entities are emitted in batches of 1000 to prevent memory issues
+      # Set to null to disable batching (only for small deployments <1k entities)
+    max_workers: 10 # Parallelize entity extraction across zones
+```
+
+**How Batching Works:**
+
+- Entries and entities are collected during API streaming
+- When a batch reaches `batch_size` entries, it's immediately emitted to DataHub
+- The batch cache is cleared to free memory
+- This keeps memory usage bounded regardless of dataset size
+- For deployments with 50k+ entities, batching prevents out-of-memory errors
+
+**Lineage Limitations:**
+
+- Dataplex does not support column-level lineage extraction
+- Lineage retention period: 30 days (Dataplex limitation)
+- Cross-region lineage is not supported by Dataplex
+- Lineage is only available for entities with active lineage tracking enabled
+  For more details on lineage limitations, refer to [GCP docs](https://docs.cloud.google.com/dataplex/docs/about-data-lineage#current-feature-limitations).
+
+### Python Dependencies
+
+The connector requires the following Python packages:
+
+```bash
+pip install 'google-cloud-dataplex>=1.0.0'
+pip install 'google-cloud-datacatalog-lineage==0.2.2'  # Required only if include_lineage is enabled
+```
```

Comment:
this is out of the box when using `acryl-datahub[dataplex]`, right?

do we need to mention this?

Reply:
yes, we don't need to mention them, modified documentation to just show they are installed by default for visibility purposes in https://github.com/datahub-project/datahub/pull/15379/commits/3a63d27fc665ece18361fa9bbf79d1f68af27e28.